src/Parser/DeathKnight/Blood/Modules/Features/BoneShield.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+class BoneShield extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+  };
+
+  SS_DR = 0.08;
+  BONE_SHIELD_DR = 0.16;
+
+  EXCLUDES_ABSORBS = [
```

Comment:
The DR chain for Bone Shield is actually
> raw damage > real DRs & armor > Blood Shield > Blood Mirror > Bone Shield > other absorbs

Maybe there are some other absorbs between BM and Bone Shield, but I only know about Blood Shield and Blood Mirror.
Blood Shield is the only absorb thats reducing Blood Mirrors reflect, kinda makes sense since it's mastery and has a higher priority.
But Blood Mirror is not reduced by Bone Shield, that's why it has to be 'Blood Shield > Blood Mirror > Bone Shield'.

But none of that should affect the Skeletal Shattering calculation, rolling the SS-dice for each hit could just introduce a larger differences (or no difference at all) between the actually absorbed damage and my estimate. That's why I went with the average crit chance to get something inbetween.

Reply:
You sure that is the ordering? Looking at logs it looks more like it's Bone Shield > Blood Mirror > Blood Shield:
![image](https://user-images.githubusercontent.com/6773230/37594696-438cbd78-2b76-11e8-884c-349e5520df9a.png)
If it was Blood Shield > Blood Mirror, Blood Mirror shouldn't have any absorb since Blood Shield Fully absorbs the hit.
And looking at dmg taken post boneshield (39.39M):
https://www.warcraftlogs.com/reports/xgqDyM9fkaA3Fnpd#fight=22&type=damage-taken&source=26&start=9786834&end=9794589
And Blood Mirror healing (7.88M)
https://www.warcraftlogs.com/reports/xgqDyM9fkaA3Fnpd#fight=22&type=healing&source=26&start=9786834&end=9794589
7.88 / 39.39 = 20% it looks like blood Mirror absorb is post bone shield.

If this is the case you don't have to worry about subtracting any absorbs.

About crit, since you don't actually roll the dice but just multiply by the chance, using the crit at the time of the event should be more accurate than average. It's a pretty rough guess either way, since it could technically contribute anywhere from 0-100% without us knowing. I'd suggest that you edit the text for SS to be something along the lines of "On average, Skeletal Shattering would have contributed X% of this." To make it clear that it is not by any means an accurate representation of how it actually performed, but only the expected value.

Reply:
You're right, looking into some 7.0 logs and you can see clearly that its Bone Shield > Blood Mirror > Blood Shield.
Asked in Discord and most people claimed that Bone Shield does not reduce Blood Mirror but Blood Shield does, should've double checked actual logs.
![bs](https://user-images.githubusercontent.com/29842841/37597865-e466510a-2b80-11e8-8f7b-a5ba9795ed18.PNG)